### PR TITLE
torchaudio RNNT loss integration test

### PR DESCRIPTION
**Pull request by carolineechen**
`test/test_e2e_asr_transducer.py::test_pytorch_transducer_gpu_trainable[torchaudio] PASSED`

### CODE REVIEW

File: test/test_e2e_asr_transducer.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def test_pytorch_transducer_trainable_and_decodable(train_dic, recog_dic):
 
 
 @pytest.mark.skipif(not torch.cuda.is_available(), reason="gpu required")
-@pytest.mark.parametrize("trans_type", ["warp-transducer", "warp-rnnt"])
+@pytest.mark.parametrize("trans_type", ["warp-transducer", "warp-rnnt", "torchaudio"])
```
Comment (carolineechen) [inline test/test_e2e_asr_transducer.py:252]: Currently torchaudio throws the following error:
```
Fatal Python error: Aborted

Current thread 0x00007fd7645d6740 (most recent call first):
  File "/data/home/carolinechen/espnet/tools/venv/lib/python3.8/site-packages/torchaudio/prototype/rnnt_loss.py", line 103 in forward
  File "/data/home/carolinechen/espnet/tools/venv/lib/python3.8/site-packages/torchaudio/prototype/rnnt_loss.py", line 169 in rnnt_loss
  File "/data/home/carolinechen/espnet/tools/venv/lib/python3.8/site-packages/torchaudio/prototype/rnnt_loss.py", line 224 in forward
  File "/data/home/carolinechen/espnet/tools/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1051 in _call_impl
  File "/data/home/carolinechen/espnet/espnet/nets/pytorch_backend/transducer/loss.py", line 81 in forward
  File "/data/home/carolinechen/espnet/tools/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1051 in _call_impl
  File "/data/home/carolinechen/espnet/espnet/nets/pytorch_backend/e2e_asr_transducer.py", line 427 in forward
  File "/data/home/carolinechen/espnet/tools/venv/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1051 in _call_impl
...
```

**Comment by vincentqb** [comment]
Verifying one more thing on the way would be important to make sure all is indeed integrated correctly: the [first commit](https://github.com/vincentqb/espnet/pull/1/commits/78c5b578b10d8070086c634c13d263b1ebad34a9) of vincentqb/espnet#1 simply runs torchaudio's along side the other tests and compares the losses. The output is shown in [comment](https://github.com/vincentqb/espnet/pull/1#issuecomment-750947324). It might be clearer to make a separate PR for this.

File: test/test_e2e_asr_transducer.py (hunk 1 above)
Comment (vincentqb) [reply test/test_e2e_asr_transducer.py:252]: Do you know the exact line where this error got triggered?

File: espnet/nets/pytorch_backend/transducer/loss.py [hunk 2]
```diff
@@ -73,6 +76,9 @@ def forward(self, pred_pad, target, pred_len, target_len):
             )
         else:
             loss = self.trans_loss(pred_pad, target, pred_len, target_len)
+            if self.trans_type == "torchaudio":
+                loss = loss.mean()
+
```
Comment (vincentqb) [inline espnet/nets/pytorch_backend/transducer/loss.py:81]: loss functions in pytorch usually have a [`reduction`](https://pytorch.org/docs/stable/nn.html#loss-functions) parameter which default to 'mean'. we should add that to our TODO :)
